OpenRA.Game/TranslationExts.cs

```diff
@@ -0,0 +1,45 @@
+#region Copyright & License Information
+/*
+ * Copyright 2007-2022 The OpenRA Developers (see AUTHORS)
+ * This file is part of OpenRA, which is free software. It is made
+ * available to you under the terms of the GNU General Public License
+ * as published by the Free Software Foundation, either version 3 of
+ * the License, or (at your option) any later version. For more
+ * information, see COPYING.
+ */
+#endregion
+
+using Linguini.Shared.Types.Bundle;
+
+namespace OpenRA
+{
+	public static class TranslationExts
+	{
+		public static IFluentType ToFluentType(this object value)
+		{
+			switch (value)
+			{
+				case byte number:
+					return (FluentNumber)number;
+				case sbyte number:
+					return (FluentNumber)number;
+				case short number:
+					return (FluentNumber)number;
+				case uint number:
+					return (FluentNumber)number;
+				case int number:
+					return (FluentNumber)number;
+				case long number:
+					return (FluentNumber)number;
+				case ulong number:
+					return (FluentNumber)number;
+				case float number:
+					return (FluentNumber)number;
+				case double number:
+					return (FluentNumber)number;
```

Comment:
Is it possible to combine these?
```suggestion
				case byte number:
				case sbyte number:
				case short number:
				case uint number:
				case int number:
				case long number:
				case ulong number:
				case float number:
				case double number:
					return (FluentNumber)number;
```

Reply:
No
> A local variable or function named 'number' is already defined in this scope [OpenRA.Game]csharp(CS0128)

Reply:
Ok, I see.

Reply:
Hm, if this is a common use case I could add something like `public static bool TryFluentNumber(this object input, out FluentNumber? operands)`

Edit or do something like [SS14](https://github.com/space-wizards/RobustToolbox/blob/dfa2c222d1d743f1acf74e60df99b0489b001096/Robust.Shared/Localization/LocalizationManager.Functions.cs#L254-L276)

Reply:
That won't work as I am restricted by https://github.com/OpenRA/OpenRA/blob/a152bf7324cbaf8656fa1e93ba96b76e4d109642/Directory.Build.props#L8